layer.go

```diff
@@ -75,7 +74,7 @@ func (l GUE) NextLayerType() gopacket.LayerType {
 }
 
 func DecodeGUE(data []byte, p gopacket.PacketBuilder) error {
-	l := GUE{}
+	l := &GUE{}
```

Comment:
stay consistent with other decoders, to avoid confusion. ref: https://github.com/google/gopacket/blob/3ed392865dcc6faee63c38c27a1f116ff85f30df/layers/ip4.go#L275